AGPs/AGP-1H1.md

```diff
@@ -0,0 +1,44 @@
+---
+AGP: tbd
+Title: Conviction Voting + Template
+Author: @lkngtn (@1hive)
+Status: Stage II
+Track: Finance
+Created: 2019-11-12
+---
+
+# **AGP-TBD: Conviction Voting + Template**
+
+## **Address of the transfer recipient**
+
+ `0x41ecc492719a9cd750f8c9a693896d0dd621d4ec` Vault of the [conviction project organization](https://mainnet.aragon.org/#/conviction
+
+## **Amount of the transfer**
+
+75,000 DAI Due upon approval
+25,000 ANT Due upon successful delivery
+
+## **Number and frequency of transfers if recurring**
+
+2 (One upon approval, one upon successful delivery).
+For the purpose of budgeting, the full amount should be debited against this quarters budget.
+
+## **Purpose of the transfer**
+
+We believe that one of most significant blockers of wider adoption of Aragon is the lack of truly public, permissionless organizations. Organizations that anyone can join and participate in immediately without asking, while at the same time mitigating known deficiencies with participation and capture associated with fund allocation in public communities.
+
+In exchange for funding we will deliver the following as project deliverables:
+* We will complete our implementation of Conviction Voting
+* We will add a locking mechanism for token manager and token wrapper
+* We will create an Escrow app that will be compatible with Aragon Court and plug in to Conviction Voting's proposal creation flow
+* We will Provide an organization template for public, permisionless organization's using Conviction Voting, Escrow, Fundraising, and Dandelion Voting
```

Comment:
> Escrow freezes the requested tokens on a Conviction Voting proposal until the review passes 

Wouldn't need to be that tightly integrated, the integration would essentially be that conviction voting proposals would either be a direct transfer (effectively how it works now), or they would instead transfer funds into an escrow contract with a pre-determined reviewer, recipient, and arbitrator. 

So imagine the following case. I want to propose performing some job for the organization, I specify the amount required for me to start work, as well as some additional party responsible for validating my work (presumably someone with some clout in the community), as well as an arbitrator in the event a dispute arises between myself and the reviewer. 

The conviction funding would happen the same as before, there would be threshold of conviction required based on the amount of funds requested relative to what is available, but the amount of requested funds may change depending on whether funds are added to the escrow contract from another source. So if there is a proposal on the table that doesn't have enough conviction to pass, but I personally want it to go through, I can deposit my own money into the escrow account (reducing the amount required to reach the goal), and then immediately lower the threshold so funds can be released from conviction voting. 

Once funds have been released from conviction voting the proposal has passed and there is no more interactions with the conviction voting app.  The recipient can request funds to be released from escrow at the discretion of the reviewer, and the reviewer can request funds to be returned proportionally to senders. If both reviewer and recipient agree great, if a dispute arises the arbitrator gets involved. 

From a practical perspective the idea behind the escrow app is to make accountability and execution simpler. We could decide that it is too much of an unknown to include here, it's a completely separate application flow. However, as a template it provides an end to end flow for how funds flow in (fundraising), and out (conviction voting + escrow). 

> Fundraising provides a market for the Conviction Voting staking and request tokens.
What is the purpose of the Dandelion Voting in this set up?

Conviction Voting effectively provides a process for fund allocation. It effectively allows fund requests and maintains a budget in terms of how quickly funds flow out of the organization. It isn't suitable for general decision making though, so if we want the organization to be able to manage upgrades then we need to have a something like Dandelion voting also installed. 

I would advocate for Dandelion Voting over the normal voting app because we can impose a delay on execution, and we can restrict people from exiting (using the fundraising bonding curve) if they vote yes on a pending proposal. This makes it more costly to vote maliciously.

Reply:
Thanks for the explanations, I understand much better now.

* Regarding the escrow app, I think it is a great idea. I'm still not sure if it should be a vault with steroids (something between a vault and an agent), or if it should be an app such as finance (connected to the DAO's vault). Also I'm not sure what is the best way to have one address per proposal:
  - I think the best practice would be to install one escrow instance per funding proposal, but I think it is overkilling: conviction voting should have Kernel and ACL permissions to install apps. I would not do that.
  - On the other hand, having just one app (being a vault with steroids or something such as finance) with a `deposit` function that specifies the proposal the sender wants to fund can be problematic for people who wants to transfer money directly to an address.
  - I think we could do something in between: a single escrow app that instantiates a (non-aragon-app non-upgradable) contract per proposal with it's own address. Those contracts just transfer the funds they receive to the `deposit` escrow function. As there is not much attack-surface and they only funnel funds, I would feel comfortable having a compromise such that. I'm eager to know your thoughts on that, @lkngtn and @fabriziovigevani.
* Regarding the Dandelion Voting app, I'd like to check it out first, but by the way I think it's a minor thing to say what voting app the template will have, and I would not specify it in the proposal.